files/requirements.dev.txt

```diff
@@ -0,0 +1,5 @@
+ansible-lint==6.15.0
```

Comment:
why do we fix the version instead of specifying the minimum allowed version? 

The fact is that everything is developing rapidly and very soon these versions will be irrelevant and we will have to periodically maintain this file so that it is up to date.

Reply:
You're right. To help manage dependencies and keep them up-to-date, developers can use automated tools like Renovate or Dependabot. These tools monitor the dependencies specified in your project and automatically create pull requests to update the versions when new releases become available. By integrating these tools into your development workflow, you can achieve the following benefits:

1. Up-to-date dependencies: Renovate and Dependabot help you stay current with the latest versions of your dependencies, ensuring you receive the benefits of new features, performance improvements, and security patches.

2. Reduced maintenance burden: Automating the update process can save developers time and effort spent on maintaining the dependencies manually. The tools handle the updates, allowing the developers to focus on other aspects of the project.

3. Compatibility checks: When a new dependency version is available, these tools create a pull request with the updated version, triggering your CI/CD pipeline. This process helps catch potential compatibility issues or conflicts with the new version before merging the changes into your main codebase.

4. Customizable configuration: You can configure the update frequency, versioning strategy (e.g., semantic versioning or pinned versions), and other settings to suit your project's requirements.

By combining the use of Renovate or Dependabot with specifying minimum allowed versions or version ranges in your requirements file, you can enjoy the advantages of both up-to-date dependencies and automated maintenance while reducing the risk of introducing breaking changes or compatibility issues.

Reply:
or just use the latest versions in tests. Or do you think that this approach is wrong?

Reply:
Based on my experience with linters, if we do not freeze their versions, they can cause errors either during the automatic CI pipelines or while running tests. This can be quite disruptive for many of my colleagues when they start working on a specific task (let's say adding a new feature) and then find themselves dealing with updates from the linters.

That's why we choose to fix the versions of all the tools we use (such as Ansible, ansible-lint, yamllint, etc.) and create specific issues to address their updates.

As updating these tools can be tedious, we've automated the process using Renovate, which gradually updates the versions while running all the tests.

This way, we can focus on addressing only the non-trivial (blocking) cases, making our workflow more efficient and less frustrating for the team.

Reply:
I've addressed the issue with the molecule version in the Makefile. That's why it's best to pin versions and update them gradually as needed.

Additionally, having a more unified approach for testing is beneficial.

I'm adding all of this here.

I know I'm advocating for my own perspective, but in any case, you remain the authority in validating this PR, so please don't hesitate to provide feedback. I don't want to come across as overly intrusive!

Reply:
OK I agree with you